data/creaturescripts/scripts/player_death.lua

```diff
@@ -59,28 +59,54 @@ function onDeath(player, corpse, killer, mostDamageKiller, lastHitUnjustified, m
 	end
 
 	local limit = deathRecords - maxDeathRecords
-	if limit > 0 then
+	if deleteRecords and limit > 0 then
 		db.asyncQuery("DELETE FROM `player_deaths` WHERE `player_id` = " .. playerGuid .. " ORDER BY `time` LIMIT " .. limit)
 	end
 
-	if byPlayer then
-		local targetGuild = player:getGuild()
-		targetGuild = targetGuild and targetGuild:getId() or 0
-		if targetGuild ~= 0 then
-			local killerGuild = killer:getGuild()
-			killerGuild = killerGuild and killerGuild:getId() or 0
-			if killerGuild ~= 0 and targetGuild ~= killerGuild and isInWar(playerId, killer:getId()) then
-				local warId = false
-				resultId = db.storeQuery("SELECT `id` FROM `guild_wars` WHERE `status` = 1 AND ((`guild1` = " .. killerGuild .. " AND `guild2` = " .. targetGuild .. ") OR (`guild1` = " .. targetGuild .. " AND `guild2` = " .. killerGuild .. "))")
-				if resultId then
-					warId = result.getNumber(resultId, "id")
-					result.free(resultId)
-				end
-
-				if warId then
-					db.asyncQuery("INSERT INTO `guildwar_kills` (`killer`, `target`, `killerguild`, `targetguild`, `time`, `warid`) VALUES (" .. db.escapeString(killerName) .. ", " .. db.escapeString(player:getName()) .. ", " .. killerGuild .. ", " .. targetGuild .. ", " .. os.time() .. ", " .. warId .. ")")
-				end
-			end
-		end
+	if not killedByPlayer then
+		return
+	end
+
+	local targetGuild = player:getGuild()
+	targetGuild = targetGuild and targetGuild:getId() or 0
+	if targetGuild == 0 then
+		return
+	end
+
+	local killerGuild = killer:getGuild()
+	killerGuild = killerGuild and killerGuild:getId() or 0
+	if killerGuild ~= 0 then
+		return
+	end
+
+	if targetGuild == killerGuild then
+		return
 	end
+
+	if not player:isAtWarAgainst(killer) then
+		return
+	end
+
+	local warId = false
+	resultId = db.storeQuery("SELECT `id` FROM `guild_wars` WHERE `status` = 1 AND ((`guild1` = " .. killerGuild .. " AND `guild2` = " .. targetGuild .. ") OR (`guild1` = " .. targetGuild .. " AND `guild2` = " .. killerGuild .. "))")
+	if resultId then
+		warId = result.getNumber(resultId, "id")
+		result.free(resultId)
+	end
+
+	if warId then
+		db.asyncQuery("INSERT INTO `guildwar_kills` (`killer`, `target`, `killerguild`, `targetguild`, `time`, `warid`) VALUES (" .. db.escapeString(killerName) .. ", " .. db.escapeString(player:getName()) .. ", " .. killerGuild .. ", " .. targetGuild .. ", " .. os.time() .. ", " .. warId .. ")")
+	end
+end
+
+function onDeath(player, corpse, killer, mostDamageKiller, lastHitUnjustified, mostDamageUnjustified)
+	player:sendTextMessage(MESSAGE_EVENT_ADVANCE, "You are dead.")
+
+	local killedByPlayer, killerName = getKiller(killer)
+	local killedByPlayerMostDamage, killerNameMostDamage = getKiller(mostDamageKiller)
+
+	takeScreenShot(killedByPlayer, mostDamageKiller)
```

Comment:
`takeScreenshot`

"screen shot" is putting a bullet through your screen

Reply:
ops